src/main/java/it/gov/pagopa/reward/notification/service/iban/outcome/IbanOutcomeOperationsServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package it.gov.pagopa.reward.notification.service.iban.outcome;
+
+import it.gov.pagopa.reward.notification.dto.iban.IbanOutcomeDTO;
+import it.gov.pagopa.reward.notification.model.RewardIban;
+import it.gov.pagopa.reward.notification.service.iban.RewardIbanService;
+import it.gov.pagopa.reward.notification.service.utils.IbanConstants;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Mono;
+
+@Service
+@Slf4j
+public class IbanOutcomeOperationsServiceImpl implements IbanOutcomeOperationsService{
+    private final RewardIbanService rewardIbanService;
+
+    public IbanOutcomeOperationsServiceImpl(RewardIbanService rewardIbanService) {
+        this.rewardIbanService = rewardIbanService;
+    }
+
+    @Override
+    public Mono<RewardIban> execute(IbanOutcomeDTO ibanOutcomeDTO) {
+        return switch (ibanOutcomeDTO.getStatus()){
+            case IbanConstants.STATUS_KO -> rewardIbanService.deleteIban(ibanOutcomeDTO);
+            case IbanConstants.STATUS_UNKNOWN_PSP -> rewardIbanService.updateStatus(ibanOutcomeDTO);
+            default -> invalidStatusType(ibanOutcomeDTO);
```

Comment:
let's change this in order to delete when KO
and update otherwise: this way we will still store the iban, but updating the outcome